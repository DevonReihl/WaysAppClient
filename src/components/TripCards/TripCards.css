.TripCard {
  height: 400px;
  width: 315px;
  border-radius: 6px;
  border: 2px rgb(104, 104, 104) solid;
  overflow: hidden;
  text-align: center;
  display: flex;
  flex-direction: column;
  background-color: white;
  margin: auto;
  margin-top: 25px;
}

.TripCard:hover {
  box-shadow: 2px 11px 11px rgb(122, 122, 122);
  border: 2px rgb(219, 219, 219) solid;
}

.TripCard-topimage {
  border-radius: 6px;
  width: 315px;
  height: 165px;
}

.TripCard-middle-section {
  height: 151px;
  display: inline-flex;
  flex-direction: column;
  justify-content: space-around;
  word-break: break-all;
  word-wrap: break-word;
}

.TripCard-title {
  color: black;
  font-weight: bold;
  font-family: 'Fredoka One', Arial, Helvetica, sans-serif;
  word-wrap: break-word;
  margin-top: 3px;
}

.TripCard-title > h2 {
  font-size: 28px;
  font-weight: 500;
  margin: 0;
  padding: 0;
}

.TripCard-topimage > img {
  height: 165px;
  width: 315px;
}

.Activities {
  font-weight: bold;
  font-size: 17px;
  min-height: 18px;
  padding-left: 1px;
  word-wrap: break-word;
}

.Activities > p {
  margin: 0;
}

.TripCard-title > span {
  color: rgb(70, 70, 70);
  font-size: 17px;
  font-family: 'Ubuntu', Arial, Helvetica, sans-serif;
}

.TripCard-bottom {
  height: 80px;
  width: 315px;
  color: rgb(8, 8, 8);
  font-family: 'Fredoka One', Arial, Helvetica, sans-serif;
  font-size: 20px;
}

.pink {
  background-color: rgb(211, 62, 136);
}

.blue {
  background-color: rgb(91, 114, 241);
}

.orange {
  background-color: rgb(233, 159, 64);
}

.violet {
  background-color: rgb(163, 83, 216);
}

.baby-blue {
  background-color: rgb(68, 157, 230);
}

.TripCard-bottom-info > span {
  font-size: 20px;
  vertical-align: middle;
}

.TripCard-bottom-info {
  padding-top: 25px;
  width: 315px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.TripCard-bottom-info > b {
  color: rgb(0, 0, 0);
  font-weight: 100;
  font-size: 18px;
  text-shadow: none;
}

img.card-heart {
  width: 2.4em;
  margin-top: 3px;
}

span.rating-span {
  display: flex;
  align-items: center;
  padding-top: 3px;
}

.empty-heart {
  padding-top: 5px;
}

@media all and (min-width: 800px) {
  .TripCard {
    margin-bottom: 0;
    margin-top: 40px;
    margin-left: 25px;
    margin-right: 25px;
  }
}
