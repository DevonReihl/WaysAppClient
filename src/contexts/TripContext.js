import React from 'react';
import UserContext from './UserContext';

const TripContext = React.createContext({
  trips: [],
  currTripId: null,
  setCurrTripId: () => {},
  setTrips: () => {},
  verifyAuth: () => {},
  setLoading: () => {},
  returnUserID: () => {},
  setTripRating: () => {},
  loading: null,
});

export default TripContext;

export class TripProvider extends React.Component {
  state = {
    trips: [],
    currTripId: null,
    currPlace: {},
    stops: [],
    loading: false,
  };

  static contextType = UserContext;

  setTrips = (res) => {
    this.setState({
      trips: res,
    });
  };

  setTripRating = (trip_id) => {
    const index = this.state.trips.findIndex((trip) => {
      return trip.id === Number(trip_id);
    });
    let trips = [...this.state.trips];
    trips[index] = {
      ...trips[index],
      rating: Number(trips[index].rating) + 1,
    };
    this.setState({ trips });
  };

  setLoading = (boolean) => {
    this.setState({
      loading: boolean,
    });
  };

  setCurrTripId = (id) => {
    this.setState({
      currTripId: id,
    });
  };

  handleAddTrip = (trip) => {
    this.setState({
      trips: [...this.state.trips, trip],
    });
  };

  verifyAuth = (id) => {
    return id === this.context.user.id;
  };

  returnUserID = () => {
    return this.context.user.id;
  };

  render() {
    const value = {
      trips: this.state.trips,
      currTripId: this.state.currTripId,
      setTrips: this.setTrips,
      setCurrTripId: this.setCurrTripId,
      verifyAuth: this.verifyAuth,
      loading: this.state.loading,
      setLoading: this.setLoading,
      returnUserID: this.returnUserID,
      setTripRating: this.setTripRating,
    };

    return (
      <TripContext.Provider value={value}>
        {this.props.children}
      </TripContext.Provider>
    );
  }
}
