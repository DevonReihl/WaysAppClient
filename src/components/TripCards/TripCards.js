import React, { Component } from 'react';
import images from '../../assets/images/images';
import { Link } from 'react-router-dom';

import './TripCards.css';

let sequence = 0;

export default class TripCards extends Component {
  sequenceColorOnCard = () => {
    let colors = ['blue', 'pink', 'orange', 'violet', 'baby-blue'];
    if (sequence > colors.length - 1) {
      sequence = 0;
    }
    let number = sequence;
    sequence++;
    return colors[number];
  };

  componentWillUnmount() {
    sequence = 0;
  }

  componentDidUpdate() {
    sequence = 0;
  }

  shouldComponentUpdate() {
    sequence = 0;
    return false;
  }

  shortifyDestination = (dest) => {
    dest = dest.split(',');
    return dest[0];
  };

  render() {
    let color = this.sequenceColorOnCard();
    return (
      <div className="TripCard">
        <Link to={`/trips/${this.props.id}`}>
          <div>
            <div className="TripCard-topimage">
              <img src={images[this.props.image]} alt="city skyline"></img>
            </div>
            <div className="TripCard-middle-section">
              <div className="TripCard-title">
                {this.props.destination.length > 32 ? (
                  <h2>{this.shortifyDestination(this.props.destination)}</h2>
                ) : (
                  <h2>{this.props.destination}</h2>
                )}
                <span>{this.props.short_description}</span>
              </div>

              <div className="Activities">
                <p>
                  Activities: <br />
                  {this.props.activities}
                </p>
              </div>
            </div>
            <div className={`TripCard-bottom ${color}`}>
              <div className="TripCard-bottom-info">
                Days {this.props.days}
                <b>&nbsp;&nbsp;|</b>
                {this.props.rating ? (
                  <span className="rating-span">
                    <img
                      src={images.FilledHeart}
                      alt="filled hearts"
                      className="card-heart"
                    />{' '}
                    {this.props.rating}
                  </span>
                ) : (
                  <span className="rating-null">
                    <img
                      src={images.EmptyHeart}
                      alt="empty heart"
                      className="card-heart empty-heart"
                    />
                  </span>
                )}
              </div>
            </div>
          </div>
        </Link>
      </div>
    );
  }
}
