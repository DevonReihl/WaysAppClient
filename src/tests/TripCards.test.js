import React from 'react'
import TripCards from '../components/TripCards/TripCards'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import TripContext from '../contexts/TripContext'

describe(`TripCards component`, () => {
    it('Renders without crashing', () => {
        shallow(
            <TripContext.Provider>
                <TripCards />
            </TripContext.Provider>
        )
    })
    it('Matches snapshot', () => {
        const wrapper =  shallow(
        <TripContext.Provider>
        <TripCards />
        </TripContext.Provider>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})