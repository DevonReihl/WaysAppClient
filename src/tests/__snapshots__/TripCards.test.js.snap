// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripCards component Matches snapshot 1`] = `<TripCards />`;
